src/main/java/com/aws/greengrass/secretmanager/SecretManager.java

```diff
@@ -58,7 +58,7 @@ public class SecretManager {
     private static final String LATEST_LABEL = "AWSCURRENT";
     public static final String VALID_SECRET_ARN_PATTERN =
             "arn:([^:]+):secretsmanager:[a-z0-9\\-]+:[0-9]{12}:secret:([a-zA-Z0-9\\\\]+/)*"
-                    + "[a-zA-Z0-9/_+=,.@\\-]+-[a-zA-Z0-9]+";
```

Comment:
just make the ending optional.

`(-[a-zA-Z0-9]+)?`

Reply:
It's not required tho. The previous word will take care of both partial and full arns.

Reply:
Making the ending optional is more correct because the change you've made allows more types of characters in the ending (eg. @)